helix-term/src/commands.rs

```diff
@@ -5209,3 +5214,124 @@ fn replay_macro(cx: &mut Context) {
         cx.editor.macro_replaying.pop();
     }));
 }
+
+fn jump_mode_word(cx: &mut Context) {
+    let jump_targets = jump::find_all_alphanumeric_word_start(cx);
+    jump_with_targets(cx, jump_targets);
+}
+
+fn jump_mode_search(cx: &mut Context) {
+    jump_mode_search_impl(cx, false);
+}
+
+fn extend_jump_mode_search(cx: &mut Context) {
+    jump_mode_search_impl(cx, true);
+}
+
+fn jump_mode_search_impl(cx: &mut Context, _extend: bool) {
+    cx.on_next_key(move |cx, event| {
+        let c = match event.char() {
+            Some(c) => c,
+            _ => return,
+        };
+
+        let (view, doc) = current!(cx.editor);
+
+        let text = doc.text().slice(..);
+        let cursor = {
+            let range = doc.selection(view.id).primary();
+            range.cursor(text)
+        };
+
+        let mut forward_jump_locations = Vec::new();
+        for n in 1.. {
+            let next = search::find_nth_next(text, c, cursor + 1, n);
+            match next {
+                Some(pos) => {
+                    let row = visual_coords_at_pos(doc.text().slice(..), pos, doc.tab_width()).row;
+                    if row >= view.offset.row + view.inner_area(doc).height as usize {
+                        break;
+                    }
+                    if !view.is_cursor_in_view(pos, doc, 0) {
+                        continue;
+                    }
+                    forward_jump_locations.push(pos);
+                }
+                _ => break,
+            }
+        }
+        let mut backward_jump_locations = Vec::new();
+        for n in 1.. {
+            let next = search::find_nth_prev(text, c, cursor, n);
+            match next {
+                Some(pos) => {
+                    let row = visual_coords_at_pos(doc.text().slice(..), pos, doc.tab_width()).row;
+                    if row < view.offset.row {
+                        break;
+                    }
+                    if !view.is_cursor_in_view(pos, doc, 0) {
+                        continue;
+                    }
+                    backward_jump_locations.push(pos);
+                }
+                _ => break,
+            }
+        }
+
+        let jump_locations: Vec<_> = forward_jump_locations
+            .into_iter()
+            .zip(backward_jump_locations.into_iter())
+            .flat_map(|(a, b)| vec![a, b])
```

Comment:
The `into_iter` and vec creation here shouldn't be necessary.
```suggestion
            .zip(backward_jump_locations)
            .flat_map(|(a, b)| [a, b])
```